espresso-support/demo/src/androidTest/java/com/novoda/movies/MovieItemViewTest.java

```diff
@@ -0,0 +1,140 @@
+import static android.support.test.espresso.matcher.ViewMatchers.withClassName;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static org.hamcrest.CoreMatchers.is;
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class MovieItemViewTest {
+
+    private static final Movie EDWARD_SCISSORHANDS = new Movie("Edward Scissorhands");
+    private static final Movie NOT_EDWARD_SCISSORHANDS = new Movie("NOT Edward Scissorhands");
+
+    @Rule
+    public ViewActivityRule<MovieItemView> viewActivityRule = new ViewActivityRule<>(R.layout.test_movie_item_view);
+
+    private HitCountingMovieItemViewListener hitCountingListener;
+
+    @Before
+    public void setUp() {
+        MovieItemView view = viewActivityRule.getView();
+
+        hitCountingListener = new HitCountingMovieItemViewListener();
+        view.attachListener(hitCountingListener);
+
+        // setup standard 'given' used for all tests
+        viewActivityRule.bindViewUsing(new ViewActivityRule.Binder<MovieItemView>() {
+            @Override
+            public void bind(MovieItemView view) {
+                view.bind(EDWARD_SCISSORHANDS);
+            }
+        });
+    }
+
+    @After
+    public void tearDown() {
+        viewActivityRule.getView().detachListeners();
+    }
+
+    @Test
+    public void clickMovieItemView() {
+        onView(withClassName(is(MovieItemView.class.getName()))).perform(click());
+
+        hitCountingListener.assertClick(EDWARD_SCISSORHANDS);
+    }
+
+    @Test
+    public void clickMovieFavoriteView() {
+        onView(withId(R.id.movie_item_button_favorite)).perform(click());
+
+        hitCountingListener.assertClickFavourite(EDWARD_SCISSORHANDS);
+    }
+
+    @Test
+    public void clickMoviePlayView() {
+        onView(withId(R.id.movie_item_button_play)).perform(click());
+
+        hitCountingListener.assertClickPlay(EDWARD_SCISSORHANDS);
+    }
+
+    @Test
+    public void rebindWithDifferentMovie_noClickMoviePlayView() {
+        // setup 'given' in test
+        viewActivityRule.bindViewUsing(new ViewActivityRule.Binder<MovieItemView>() {
+            @Override
+            public void bind(MovieItemView view) {
+                view.bind(NOT_EDWARD_SCISSORHANDS);
+            }
+        });
+
+        onView(withId(R.id.movie_item_button_play)).perform(click());
+
+        // like `Mockito.verify(foo, never())`
+        hitCountingListener.assertNoClickPlay(EDWARD_SCISSORHANDS);
+    }
+
+    private static class HitCountingMovieItemViewListener implements MovieItemView.Listener {
+
+        private static final String ON_CLICK = "onClick ";
+        private static final String ON_CLICK_PLAY = "onClickPlay ";
+        private static final String ON_CLICK_FAVORITE = "onClickFavorite ";
+
+        private final HitCounter hitCounter = new HitCounter();
+
+        @Override
+        public void onClick(Movie movie) {
+            hitCounter.markHit(ON_CLICK + movie.name);
```

Comment:
the `HitCounter` is a wrapper around a `Map<CharSequence,Boolean>`. This means you can be specific with your asserts by generating keys as a combination of method calls and method params as we do here.